sql/parser/builtins.go

```diff
@@ -943,6 +961,89 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+func regexpReplace(s, pattern, to, flags string) (Datum, error) {
+	pattern, global := regexpEvalFlags(pattern, flags)
+
+	patternRe, err := regexp.Compile(pattern)
+	if err != nil {
+		return DNull, err
+	}
+
+	matchCount := 1
+	if global {
+		matchCount = -1
+	}
+
+	replaceIndex := 0
+	var newString bytes.Buffer
+	for _, matchIndex := range patternRe.FindAllStringSubmatchIndex(s, matchCount) {
```

Comment:
`ReplaceAllStringFunc` is what I initially used to try to implement this, but I ran into the issue that it didn't expose regexp submatches during replacement, which is needed for submatch substitution (eg. `"\1"`).

I then tried to use `ReplaceAllString`, which does allow for submatch expansion, but that didn't allow for limiting replacements or entire match substitution (eg. `"\&"`).